### PR TITLE
Stats: Deprecating old Stats experience

### PR DESCRIPTION
**Pull request by grzegorz-cp**
Fixes https://github.com/Automattic/red-team/issues/11

## Proposed changes:

* add a banner to old Stats experience informing users that it is deprecated

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

no

## Testing instructions:

* Go to Jetpack -> Settings and disable the new experience
* Go to Jetpack -> Stats and verify that the page has a banner informing that the Old Jetpack Stats is deprecated

### CODE REVIEW

File: projects/plugins/jetpack/modules/stats.php [hunk 1]
```diff
@@ -461,6 +549,21 @@ function stats_reports_page( $main_chart_only = false ) {
 			?>
 			</h2>
 		</div>
+		<div class="wrap">
+			<div class="stats-odyssey-notice">
+				<div class="stats-odyssey-notice--content">
+					<h2 class="stats-odyssey-notice--content-header"><?php esc_html_e( 'Old Jetpack Stats experience', 'jetpack' ); ?></h2>
+					<p class="stats-odyssey-notice--content-text"><?php esc_html_e( 'The old Jetpack Stats has been deprecated and will be removed soon.', 'jetpack' ); ?></p>
+					<div class="stats-odyssey-notice--action-bar">
+						<button class="dops-button stats-odyssey-notice--primary-button">
```
Comment (grzegorz-cp) [inline projects/plugins/jetpack/modules/stats.php:563]: The button enables the new experience and doesn't require users to visit the settings page.

File: projects/plugins/jetpack/modules/stats.php [hunk 2]
```diff
@@ -423,14 +423,102 @@ function stats_reports_page( $main_chart_only = false ) {
...
+			line-height: 32px;
+			margin: 0;
+			margin-bottom: 8px;
+		}
+		.stats-odyssey-notice--content-text {
+			font-size: 16px;
+			margin: 0;
+		}
+		.stats-odyssey-notice--image-container {
+			background-image: url("/wp-content/plugins/jetpack/images/odyssey-upgrade/background.png"), url("/wp-content/plugins/jetpack/images/odyssey-upgrade/gradient.png");
```
Comment (kangzj) [inline projects/plugins/jetpack/modules/stats.php]: the URL of the image could change when wp is placed in a sub directory. Sometimes, wp-content could be another name etc. Could we please change it to use a function to get the URL, for example `wp_plugin_dir()`.

**Review by kangzj — COMMENTED** [review]
It generally works well for me. We might want to address the inline comment before merging.

Another suggestion, could we make it visually stronger? So instead of the black color for normal notices, could we use orange or red? I vaguely remember they have a warning/error style we could possibly leverage. I guess you could refer to the styling here: https://github.com/Automattic/wp-calypso/issues/84671

Thank you so much for your fast work on this 👍

**Comment by grzegorz-cp** [comment]
I changed the border to red and I experimented with setting the front to red as well. I think it doesn't look good and isn't consistent with other banners. Let me know if you think we should change despite this.

SCR-20240529-jefz

File: projects/plugins/jetpack/modules/stats.php [hunk 3]
```diff
@@ -461,6 +554,21 @@ function stats_reports_page( $main_chart_only = false ) {
 			?>
 			</h2>
 		</div>
+		<div class="wrap">
+			<div class="stats-odyssey-notice stats-odyssey-notice--content__highlighted">
+				<div class="stats-odyssey-notice--content">
+					<h2 class="stats-odyssey-notice--content-header"><?php esc_html_e( 'Old Jetpack Stats experience', 'jetpack' ); ?></h2>
```
Comment (kangzj) [inline projects/plugins/jetpack/modules/stats.php]: Maybe "We are deprecating the legacy Jetpack Stats experience" to clarify what the notice is about?

File: projects/plugins/jetpack/modules/stats.php [hunk 4]
```diff
@@ -461,6 +554,21 @@ function stats_reports_page( $main_chart_only = false ) {
...
+					<p class="stats-odyssey-notice--content-text"><?php esc_html_e( 'The old Jetpack Stats has been deprecated and will be removed soon.', 'jetpack' ); ?></p>
```
Comment (kangzj) [inline projects/plugins/jetpack/modules/stats.php]: Maybe "The old Jetpack Stats has been deprecated and will be removed soon. Please click the button to enable the new experience."

File: projects/plugins/jetpack/modules/stats.php (hunk 3 above)
Comment (kangzj) [inline projects/plugins/jetpack/modules/stats.php]:
```suggestion
					<h2 class="stats-odyssey-notice--content-header"><?php esc_html_e( 'We are deprecating the legacy Jetpack Stats experience', 'jetpack' ); ?></h2>
```

File: projects/plugins/jetpack/modules/stats.php (hunk 4 above)
Comment (kangzj) [inline projects/plugins/jetpack/modules/stats.php]:
```suggestion
					<p class="stats-odyssey-notice--content-text"><?php esc_html_e( 'The old Jetpack Stats has been deprecated and will be removed soon. Please click the button to enable the new experience.', 'jetpack' ); ?></p>
```

File: projects/plugins/jetpack/modules/stats.php [hunk 5]
```diff
@@ -461,6 +554,21 @@ function stats_reports_page( $main_chart_only = false ) {
 			?>
 			</h2>
 		</div>
+		<div class="wrap">
+			<div class="stats-odyssey-notice stats-odyssey-notice--content__highlighted">
+				<div class="stats-odyssey-notice--content">
+					<h2 class="stats-odyssey-notice--content-header"><?php esc_html_e( 'We are deprecating the legacy Jetpack Stats experience', 'jetpack' ); ?></h2>
```
Comment (kangzj) [inline projects/plugins/jetpack/modules/stats.php]:
```suggestion
					<h2 class="stats-odyssey-notice--content-header"><?php esc_html_e( 'Deprecated Jetpack Stats Experience', 'jetpack' ); ?></h2>
```

**Review by kangzj — APPROVED** [review]
Looks great to me! Tweaked the copy a bit in the last minute. Hope you don't mind 👍

**Pull request merged**